crates/pathfinder/src/consensus/inner/p2p_task.rs

```diff
@@ -92,17 +92,25 @@ pub fn spawn(
 
                     match event {
                         Event::Proposal(height_and_round, proposal_part) => {
-                            match handle_incoming_proposal_part(
-                                chain_id,
-                                height_and_round,
-                                proposal_part,
-                                &mut incoming_proposals_cache,
-                                &mut validator_cache,
-                                &mut deferred_executions,
-                                storage.clone(),
-                            )
-                            .await
-                            {
+                            let vcache = validator_cache.clone();
+                            let dex = deferred_executions.clone();
+                            let storage = storage.clone();
+                            let consensus_storage = consensus_storage.clone();
+                            let result = util::task::spawn_blocking(async move |_| {
+                                handle_incoming_proposal_part(
+                                    chain_id,
+                                    validator_address,
+                                    height_and_round,
+                                    proposal_part,
+                                    vcache,
+                                    dex,
+                                    storage,
+                                    consensus_storage,
+                                )
+                            })
+                            .await?
+                            .await;
```

Comment:
```suggestion
                            let result = util::task::spawn_blocking(move |_| {
                                handle_incoming_proposal_part(
                                    chain_id,
                                    validator_address,
                                    height_and_round,
                                    proposal_part,
                                    vcache,
                                    dex,
                                    storage,
                                    consensus_storage,
                                )
                            })
                            .await?;
```
Otherwise it'll still execute blocking code within async context :thinking: .